pkg/config/config.go

```diff
@@ -245,6 +247,22 @@ func GetMockACPIPowerPath() string {
 	return MockACPIPowerPath
 }
 
+func SetMachineSpecFilePath(specFilePath string) {
+	machineSpecFilePath = specFilePath
+}
+
+// GetMachineSpec initializes a map of MachineSpecValues from MACHINE_SPEC
+func GetMachineSpec() *MachineSpec {
+	if machineSpecFilePath != "" {
+		if spec, err := readMachineSpec(machineSpecFilePath); err == nil {
+			return spec
+		} else {
+			klog.Errorf("failed to read spec from %s: %v", machineSpecFilePath, err)
+		}
```

Comment:
minor: 
```suggestion
		}
		klog.Errorf("failed to read spec from %s: %v", machineSpecFilePath, err)
```

In this case, shouldn't we return an error and fail?

Reply:
Updated to warning.